src/pudl/output/ferc714.py

```diff
@@ -0,0 +1,400 @@
+
+        The FERC 714 Respondent IDs and names are reported in their own table,
+        without any refence to individual years, but much of the information we are
+        associating with them varies annually. This method creates an annualized
+        version of the respondent table, with each respondent having an entry
+        corresponding to every year in which hourly demand was reported in the FERC 714
+        dataset as a whole -- this necessarily means that many of the respondents will
+        end up having entries for years in which they reported no demand, and that's
+        fine.  They can be filtered later.
+        """
+        if update or self._annualized is None:
+            # Calculate the total demand per respondent, per year:
+            report_dates = (
+                self.pudl_out.demand_hourly_pa_ferc714()
+                .report_date.unique()
+            )
+            self._annualized = (
+                self.pudl_out.respondent_id_ferc714()
+                .pipe(add_dates, report_dates)
+                .pipe(
+                    pudl.helpers.convert_cols_dtypes,
+                    data_source="ferc714",
+                    name="FERC 714 Respondents Annualized",
+                )
+            )
+        return self._annualized
+
+    def categorize(self, update=False):
+        """
+        Annualized respondents with ``respondent_type`` assigned if possible.
+
+        Categorize each respondent as either a ``utility`` or a ``balancing_authority``
+        using the parameters stored in the instance of the class. While categorization
+        can also be done without annualizing, this function annualizes as well, since
+        we are adding the ``respondent_type`` in order to be able to compile service
+        territories for the respondent, which vary annually.
+        """
+        if update or self._categorized is None:
+            rids_ferc714 = self.pudl_out.respondent_id_ferc714()
+            categorized = (
+                pudl.analysis.demand_mapping.categorize_eia_code(
+                    rids_ferc714.eia_code.dropna().unique(),
+                    ba_ids=self.ba_ids,
+                    util_ids=self.util_ids,
+                    priority=self.priority)
+                .merge(self.annualize(update=update), how="right")
+            )
+            # Names, ids, and codes for BAs identified as FERC 714 respondents
+            # NOTE: this is not *strictly* correct, because the EIA BAs are not
+            # eternal and unchanging.  There's at least one case in which the BA
+            # associated with a given ID had a code and name change between years
+            # after it changed hands. However, not merging on report_date in
+            # addition to the balancing_authority_id_eia / eia_code fields ensures
+            # that all years are populated for all BAs, which keeps them analogous
+            # to the Utiliies in structure. Sooo.... it's fine for now.
```

Comment:
The problem here is that because the FERC 714 associates a particular EIA ID with each FERC Respondent, and the FERC Respondents *are* enternal and unchanging, there are mismatches between the two datasets -- if I leave the EIA BA IDs out in the years where they don't show up in EIA, then I end up with a collection of respondent-years on the FERC side that can't be matched to a BA.